File: Cpp/26.cpp
#include<iostream>
#include<vector>

using namespace std;

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        if(nums.size() == 0){
            return 0;
        }
        
        int i=0, start=0;
        while(i<nums.size()){
            if(nums[start] != nums[i]){
                nums[++start] = nums[i];
            }
            i++;
        }
        
        return start+1;
    }
};

int main(){
	vector<int> testcase={1,1,2};
	Solution s;
	int length=s.removeDuplicates(testcase);
	for(int i=0;i<length;++i){
		cout<<testcase[i]<<", ";
	}
	cout<<endl;
	return 0;
}

File: Python3/26.py
class Solution:
    def removeDuplicates(self, nums: List[int]) -> int:
        start = 0
        for i in range(1, len(nums)):
            # compare the value with coming new values,
            # if it is different, assign the new value to the next position.
            if nums[start] != nums[i]:
                start +=1
                nums[start] = nums[i]
        return start + 1
